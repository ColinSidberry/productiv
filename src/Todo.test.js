import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const TEST_DATA = {
  id: 1,
  title: "test",
  description: "testing data",
  priority: 1,
};

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<Todo />);
  });
  it("matches snapshot", function () {
    const { container } = render(
      <Todo
        id={TEST_DATA.id}
        title={TEST_DATA.title}
        description={TEST_DATA.description}
        priority={TEST_DATA.priority}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("contains expected title", function () {
    const result = render(
      <Todo
        id={TEST_DATA.id}
        title={TEST_DATA.title}
        description={TEST_DATA.description}
        priority={TEST_DATA.priority}
      />
    );
    expect(result.queryByText("test")).toBeInTheDocument();
    expect(result.queryByText("(priority: 1)"));
  });
});
