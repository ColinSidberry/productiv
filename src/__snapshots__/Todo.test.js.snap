// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        test
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        testing data
      </small>
    </div>
  </div>
</div>
`;
